google/cloud/bigquery/_pandas_helpers.py

```diff
@@ -854,11 +853,11 @@ def _download_table_bqstorage(
             the queue used to buffer downloaded data. If None, the queue
             is unbounded. Defaults to _MAX_QUEUE_SIZE_DEFAULT.
         max_stream_count (Optional[int]): The maximum number of
-            concurrent streams to use for downloading data. If None, the
-            number of streams is determined automatically based on the
-            `preserve_order` parameter. If `max_stream_count` is set to a
-            non-negative value it will override values for `preserve_order`.
-            Defaults to None.
+            concurrent streams to use for downloading data. If `preserve_order`
+            is True, `max_stream_count` is ignored. If `preserve_order` is False
+            and `max_stream_count` is set to a non-negative integer, that number
+            of streams will be requested. If `max_stream_count` is None, then the
+            number of requested streams will be unbounded. Defaults to None.
```

Comment:
nit: I think this could be simplified. Maybe something like below. 
```suggestion
        max_stream_count (Optional[int]): The maximum number of
            concurrent streams to use for downloading data. If `preserve_order`
            is True, the requested streams are limited to 1 regardless of the 
            `max_stream_count` value. If 0 or None, then the number of
            requested streams will be unbounded. Defaults to None.
```